dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -545,16 +545,20 @@ private boolean shouldProcessPush(GitHubSourceCodeRepo gitHubSourceCodeRepo, Str
     private boolean shouldProcessDelete(GitHubSourceCodeRepo gitHubSourceCodeRepo, String repository, String reference) {
         try {
             // Process the delete if the reference does not exist (there is no head commit).
-            return getHeadCommit(gitHubSourceCodeRepo, repository, reference) == null;
+            return getCurrentHash(gitHubSourceCodeRepo, repository, reference) == null;
         } catch (CustomWebApplicationException ex) {
             // If there's a problem determining if the delete needs processing, assume it does.
             LOG.info("CustomWebApplicationException determining whether to process delete", ex);
             return true;
         }
     }
 
-    private String getHeadCommit(GitHubSourceCodeRepo repo, String repository, String reference) {
-        return repo.getCommitID(repository, reference);
+    private String getCurrentHash(GitHubSourceCodeRepo repo, String repository, String reference) {
+        if (StringUtils.startsWith(reference, "refs/tags/")) {
```

Comment:
A comment here or as JavaDoc would be nice. I've already forgotten about annotated tags once, :) a reminder when looking at this code in the future would be nice.